docs/api-change-runbook.md

```diff
@@ -0,0 +1,264 @@
+# Runbook for Firecracker API changes
+
+This runbook will cover triaging API changes and ways to implement them
+appropriately.
+
+## Definitions
+
+* *Deprecated* - We will consider a deprecated API endpoint to be an endpoint
+    which still has backing functionality and can be used, but will soon be
+    removed completely along with said functionality in an upcoming version.
+* *Mandatory endpoint* - We will consider an endpoint mandatory if Firecracker
+    cannot operate normally without performing a request to it.
+* *Optional endpoint* - We will consider an endpoint optional if Firecracker
+    can operate normally without performing a request to it and the
+    functionality behind it is not essential.
+* *Mandatory header/field* - We will consider a header/field mandatory in an
+    HTTP message if the request will fail without specifying said header/field.
+* *Optional header/field* - We will consider a header/field optional in an
+    HTTP message if the request will succeeds without specifying said
+    header/field.
+
+## Triaging API changes
+
+For the purposes of this document, there are 2 main categories for API changes,
+namely *breaking* and *non-breaking*.
+
+### What is a breaking change?
+
+A breaking change in the API is a change that makes the API backwards
+incompatible with the previous version. A breaking change may include
+deprecation and incrementing the minor version in an effort to preserve
+backwards compatibility, but will always ultimately result in incrementing
+the major version. Here is a non-exhaustive list of such changes:
```

Comment:
I'm not sure I understood what the following means:

```
A breaking change may include
deprecation and incrementing the minor version in an effort to preserve
backwards compatibility, but will always ultimately result in incrementing
the major version.
```

Reply:
Rephrased. Let me know if it's better like this.